schemas/EGA.ArrayAssay.json

```diff
@@ -0,0 +1,89 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.ArrayAssay.json",
+    "type": "object",
+    "title": "EGA ArrayAssay metadata schema",
+    "description": "Metadata schema used by the European Genome-phenome Archive (EGA) to validate its ArrayAssay metadata object. This object is intended to contain metadata about the physical array to which the assayed molecule is hybridized. Used exclusively an Array Format (AF) submission. Further details can be found in the EGA-metadata-schema GitHub repository (https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas) and EGA-archive website (https://ega-archive.org/)",
+    "required": ["object_ids", "arrayAssay_center", "assay_files"],
+    "additionalProperties": false,
+    "properties": {
+      "object_ids": {
+        "type": "object",
+        "title": "Object's IDs block",
+        "description": "Node containing the main identifiers of the object (e.g. alias, center_name...), inherited from the common definitions. #! We extend the core object (object_coreIDs) by adding a pattern check based on this schema's nature: an ArrayAssay (by using EGA-ArrayAssay-id-pattern)", 
+        "allOf": [
+          {
+            "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/object_coreIDs"
+          },
+          {
+            "properties": {
+              "accession": {
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-ArrayAssay-id-pattern"
+              }
+            }
+          }
+        ]        
+      },
+
+      "arrayAssay_center": {
+        "type": "string",
+        "title": "Centername that performed the ArrayAssay",
+        "description": "The name of the center (e.g. 'EBI-TEST') responsible for the profiling by microarray (executed the ArrayAssay), if applicable (in case it’s different from the center submitting metadata)."
+      },
+
+      "arrayAssay_date": {
+        "type": "string",
+        "title": "Date of the ArrayAssay",
+        "description": "ISO date (format YYYY-MM-DD - e.g. '2021-05-15') when the ArrayAssay took place. If the protocols are long enough, the date shall be the day the ArrayAssay concluded.",
+        "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-ISO-date-YYYY-MM-DD-pattern"
```

Comment:
The ``$ref`` fields are pointers of other ``$id`` fields, whether they are in the same file or in another file. As far as I read, common practices regarding JSON schemas state that ``$id`` fields need to be identifiers, but not necessarily network-resolvable. I chose to use as the base of my identifiers the EGA-metadata GitHub URI, just so that nothing else on the web uses a similar identifier. 

See a related piece of documentation from the [Schemas Identification](https://json-schema.org/understanding-json-schema/structuring.html#schema-identification):

> Even though schemas are identified by URIs, those identifiers are not necessarily network-addressable. They are just identifiers.